models/oneinch/fantom/oneinch_fantom_calls_transfers.sql

```diff
@@ -0,0 +1,126 @@
+{{ config( 
+    alias = alias('fantom_calls_transfers'),
+    tags = ['dunesql'],
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['tx_hash', 'start', 'transfer_trace_address']
+    )
+}}
+
+{% set project_start_date = "timestamp '2020-11-13'" %} 
+{% set lookback_days = -2 %}
+{% set transfer_selector = '0xa9059cbb' %}
+{% set transfer_from_selector = '0x23b872dd' %}
+{% set selector = 'substr(input, 1, 4)' %}
+
+with
+
+calls as (
+    select 
+        tx_hash
+        , gr_traces.start
+        , gr_traces.caller
+        , transactions.tx_from
+        , transactions.tx_success
+        , gr_traces.call_success
+        , gr_traces.call_selector
+    from (
+        select 
+            "from" as tx_from
+            , "hash" as tx_hash
+            , success as tx_success
+            , block_time from {{ source('fantom', 'transactions') }}
+        where
+        {% if is_incremental() %}
+            block_time >= date_add('day', {{lookback_days}}, now())
+        {% else %}
+            block_time >= {{ project_start_date }}
+        {% endif %}
+    ) as transactions 
+    join (
+        select 
+            tx_hash
+            , min(trace_address) as start
+            , min_by("from", trace_address) as caller
+            , min_by(success, trace_address) as call_success
+            , min_by({{selector}}, trace_address) as call_selector
+        from {{ source('fantom', 'traces') }}
+        where 
+            {% if is_incremental() %}
+                block_time >= date_add('day', {{lookback_days}}, now())
+            {% else %}
+                block_time >= {{ project_start_date }}
+            {% endif %}
+            and "to" in (
+                select contract_address from {{ ref('oneinch_contract_addresses') }}
```

Comment:
i believe this is where you're failing the CI test. since this spell is dependent on upstream spell(s), you need to also migrate those. we should focus on most upstream first, then downstream. if this ref is a simple spell, feel free to add the migrated version to this PR.

please be sure to modify the non-legacy file -- we generated legacy versions to maintain spark logic. there should be this file without legacy suffix.

Reply:
if by "migration" of that upstream spell you mean adding tag and alias — it's done in the earlier PR waiting for merge https://github.com/duneanalytics/spellbook/pull/3784/files

(also, should we remove that workaround with `'0x...'` quotes and pass varbinary normally, without quotes? if yes i'd just do all of these things in this PR and ask to cancel 3784)

Reply:
yes exactly, that spell.

i would recommended pulling those changes into this PR, since it's small. then we can handle the few dependencies together quickly.

you can simply remove the single quotes on all the addresses, and dunesql engine will write as varbinary. also ensure you remove any functions on the address column like lower/cast/whatever else. just leave the hardcoded address as-is without quotes

Reply:
thanks! done